cli-paho-java/src/main/java/com/redhat/mqe/amc/Client.java

```diff
@@ -99,6 +109,16 @@ OptionParser populateOptionParser(OptionParser parser) {
         logMsgs = parser.accepts("log-msgs", "print messages").withRequiredArg()
             .ofType(String.class).defaultsTo("none");
 
+        willFlag = parser.accepts("will-flag", "will flag (0,1) ").withRequiredArg().ofType(Integer.class).defaultsTo(0);
+
+        willMessage = parser.accepts("will-message", "will message").withRequiredArg().ofType(String.class).defaultsTo("");
+
+        willQos = parser.accepts("will-qos", "will QoS (0,1,2) ").withRequiredArg().ofType(Integer.class).defaultsTo(0);
+
+        willRetained = parser.accepts("will-retained", "is will retained").withRequiredArg().ofType(Boolean.class).defaultsTo(false);
+
+        willDestination = parser.accepts("will-destination", "will topic name").withRequiredArg().ofType(String.class).defaultsTo("");
```

Comment:
following the naming of other options, why not --conn-will-destination, etc. The will message is related to connection, IMO.